### DISCUSSION TITLE
ArduinoOTA, ENCEthernet and STM32

### DISCUSSION THREAD

**Discussion by luca-stm32** [JAndrassy/ArduinoOTA#185] [Q&A]
Hi JAndrassy.
First of all, I woulk like to thank you for your work! I use yout ENCEthernet library with STM32 core ( vith flash > 128k) for my home projects with success.
I woul like to implement your ArduinoOTA library to upload new firmware, but I would like to flash my board not from IDE nor from server but via web page as in ESP8266HTTPUpdateServer example.
I haven't found any example like this in your library examples: is there a place where I could find it (with ENCEthernet and STM32 core)? 
I found an example that uses an external ESP used as programmer, but I prefer not to use any external micros as I already have an STM32 board connected to LAN via ENC28J60 ic.

Thanks for your time.
Regards.
Luca

**Comment by JAndrassy** [comment]
the Advanced examples show how to use InternalStorage with custom binary transport

**Reply by luca-stm32** [reply]
I already seen that examples.
I don't want you to make the code for me, but I'm not very keen on Internet protocols.  I just want to ask you if you have some kind of example where I could find some suggestions: I already seen how ESP do the job, but I haven't understood how to modify the code in order to use it in my application. Maybe for me this could be the opportunity to learn something new :-)

**Reply by JAndrassy** [reply]
so the OTA server in the library is a HTTP server so you can copy the upload handling from there
https://github.com/JAndrassy/ArduinoOTA/blob/86433734057cbd4b0c54abdb9be8d045cd55daf8/src/WiFiOTA.cpp#L230

**Reply by luca-stm32** [reply]
Thanks again for your support!
I will try to implement it in my project :-)

Best regards.
Luca
